### DISCUSSION TITLE
Generation Of Sub-Types

### DISCUSSION THREAD

**Discussion by metasophiea** [Aleph-Alpha/ts-rs#177] [Q&A] [answered]
Here there!

Lets say I have a struct like this

`SomeStruct {
    thing: usize,
    anotherThing: CoolEnum
}`

and I annotate it correctly so that the typescript bindings are generated as we would expect. "CoolEnum" also has the correct annotations, but is being imported from another crate. In this situation, I've found that the "SomeStruct.ts" file contains a line to import the "CoolEnum" type, but the "CoolEnum.ts" itself isn't generated. For that I need to go over to the crate that has "CoolEnum" and transport the typescript file myself.

I can understand why, but is there perhaps an option I'm missing somewhere where I can ask that the system generate typescript files for all the types used within a type?

**Comment by siefkenj** [comment]
I'm having this exact issue. If the types are declared in another module, even with `#[derive(TS)]`, it seems they won't export.

**Reply by NyxCode** [reply]
In an other module, or in an other crate?

**Reply by siefkenj** [reply]
Sorry, in a separate crate.

**Reply by escritorio-gustavo** [reply]
If `CoolEnum` is from another crate and doesn't implement `TS` this is not gonna work, as all fields must implement `TS` for `#[derive(TS)]` to work.

Of course, due to Rust's orphan rule you cannot do `impl TS for CoolEnum` yourself, which is unfortunate.

Thankfully a solution is being worked on! Keep an eye on #113 and its related PR #174

**Comment by escritorio-gustavo** [comment, marked as answer]
Solved as of #218
